### PR TITLE
Implement JVM_IsImplicitlyConstructibleClass

### PR DESCRIPTION
**Pull request by theresa-m**
Related to: https://github.com/eclipse-openj9/openj9/issues/19460

### CODE REVIEW

File: runtime/j9vm/javanextvmi.cpp [hunk 1]
```diff
@@ -741,8 +741,24 @@ JVM_IsValhallaEnabled()
...
+	jboolean result = JNI_FALSE;
+	J9VMThread *currentThread = (J9VMThread *)env;
+	J9InternalVMFunctions const * const vmFuncs = currentThread->javaVM->internalVMFunctions;
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+	if (NULL == cls) {
+		vmFuncs->setCurrentException(currentThread, J9VMCONSTANTPOOL_JAVALANGNULLPOINTEREXCEPTION, NULL);
+	} else {
+		J9Class *clazz = J9VM_J9CLASS_FROM_JCLASS(currentThread, cls);
+		J9ROMClass *romClass = clazz->romClass;
+		if (J9ROMCLASS_IS_VALUE(romClass)
```
Comment (hangshao0) [inline runtime/j9vm/javanextvmi.cpp:753]: Is the `J9ROMCLASS_IS_VALUE(romClass)` check needed ?

Comment (theresa-m) [reply runtime/j9vm/javanextvmi.cpp:753]: No I can remove that.

**Comment by hangshao0** [comment]
Jenkins test sanity alinuxval jdknext

**Review by hangshao0 — APPROVED** [review]
LGTM.

**Pull request merged**